src/main/minio.js

```diff
@@ -1148,6 +1148,120 @@ export class Client {
     return readStream
   }
 
+  // list a batch of objects using S3 ListObjects v2
+  listObjectsV2Query(bucketName, prefix, continuationToken, delimiter, maxKeys) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if (!isString(prefix)) {
+      throw new TypeError('prefix should be of type "string"')
+    }
+    if (!isString(continuationToken)) {
+      throw new TypeError('continuationToken should be of type "string"')
+    }
+    if (!isString(delimiter)) {
+      throw new TypeError('delimiter should be of type "string"')
+    }
+    if (!isNumber(maxKeys)) {
+      throw new TypeError('maxKeys should be of type "number"')
+    }
+    var queries = []
+
+    // Call for listing objects v2 API
+    queries.push(`list-type=2`)
+
+    // escape every value in query string, except maxKeys
+    if (prefix) {
+      prefix = uriEscape(prefix)
+      queries.push(`prefix=${prefix}`)
+    }
+    if (continuationToken) {
+      continuationToken = uriEscape(continuationToken)
+      queries.push(`continuation-token=${continuationToken}`)
+    }
+    if (delimiter) {
+      delimiter = uriEscape(delimiter)
+      queries.push(`delimiter=${delimiter}`)
+    }
+    // no need to escape maxKeys
+    if (maxKeys) {
+      if (maxKeys >= 1000) {
+        maxKeys = 1000
+      }
+      queries.push(`max-keys=${maxKeys}`)
+    }
+    queries.sort()
+    var query = ''
+    if (queries.length > 0) {
+      query = `${queries.join('&')}`
```

Comment:
You could use [`querystring#stringify()`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options) here if you'd like.